tests/unit/synapseutils/unit_test_synapseutils_sync.py

```diff
@@ -773,3 +777,160 @@ def test_get_file_entity_provenance_dict__error_not_404(self):
+    with patch.object(syn, "get", return_value=Project()):
+        sync.readManifestFile(syn, manifest)
+
+
+@patch.object(sync, 'os')
+def test_check_file_name_with_illegal_char(mock_os, syn):
+    """
+    Verify the check_file_name method raises the ValueError when the file name contains illegal char
+    """
+
+    project_id = "syn123"
+    header = 'path\tparent\tname\n'
+    path1 = os.path.abspath(os.path.expanduser('~/file1.txt'))
+    path2 = os.path.abspath(os.path.expanduser('~/file2.txt'))
+    path3 = os.path.abspath(os.path.expanduser('~/file3.txt'))
+    path4 = os.path.abspath(os.path.expanduser('~/file4.txt'))
+
+    row1 = f"{path1}\t{project_id}\tTest_file_name.txt\n"
+    row2 = f"{path2}\t{project_id}\tTest_file-name`s(1).txt\n"
+    row3 = f"{path3}\t{project_id}\t\n"
+    row4 = f"{path4}\t{project_id}\tTest_file_name_with_#.txt\n"
+
+    manifest = StringIO(header + row1 + row2 + row3 + row4)
+    mock_os.reset_mock()
+    mock_os.path.isfile.side_effect = [True, True, True, True]
+    mock_os.path.abspath.side_effect = [path1, path2, path3, path4]
+    mock_os.path.basename.return_value = 'file3.txt'
+
+    with pytest.raises(ValueError) as ve:
+        sync.readManifestFile(syn, manifest)
+    assert str(ve.value) == "The file name on your local side is invalid to store on Synapse. Names may only contain:" \
+                            "letters, numbers, spaces, underscores, hyphens, periods, plus signs, " \
+                            "apostrophes,and parentheses"
+
+
+@patch.object(sync, 'os')
+def test_check_file_name_with_too_long_filename(mock_os, syn):
+    """
+    Verify the check_file_name method raises the ValueError when the file name is too long
+    """
+
+    project_id = "syn123"
+    header = 'path\tparent\tname\n'
+    path1 = os.path.abspath(os.path.expanduser('~/file1.txt'))
+    path2 = os.path.abspath(os.path.expanduser('~/file2.txt'))
+    path3 = os.path.abspath(os.path.expanduser('~/file3.txt'))
+    path4 = os.path.abspath(os.path.expanduser('~/file4.txt'))
+
+    long_file_name = 'test_filename_too_long_test_filename_too_long_test_filename_too_long_test_filename_too_long_' \
+                     'test_filename_too_long_test_filename_too_long_test_filename_too_long_test_filename_too_long_' \
+                     'test_filename_too_long_test_filename_too_long_test_filename_too_long_test_filename_too_long_'
+
+    row1 = f"{path1}\t{project_id}\tTest_file_name.txt\n"
+    row2 = f"{path2}\t{project_id}\tTest_file-name`s(1).txt\n"
+    row3 = f"{path3}\t{project_id}\t\n"
+    row4 = f"{path4}\t{project_id}\t{long_file_name}\n"
+
+    manifest = StringIO(header + row1 + row2 + row3 + row4)
+    mock_os.reset_mock()
+    mock_os.path.isfile.side_effect = [True, True, True, True]
+    mock_os.path.abspath.side_effect = [path1, path2, path3, path4]
+    mock_os.path.basename.return_value = 'file3.txt'
```

Comment:
Why file3.txt here?

Reply:
Because we still have one name column is empty which path is path3 = os.path.abspath(os.path.expanduser('~/file3.txt'))
if the row['name'] is empty we will call os.path.basename to get the file name on the path